doc/source/reference/c-api/miscellaneous.rst

```diff
@@ -0,0 +1,841 @@
+
+.. c:function:: PyArray_XDECREF_ERR(PyObject* obj)
+
+    Deprecated in 1.14, use :c:func:`PyArray_DiscardWritebackIfCopy`
+    followed by ``Py_XDECREF``
+
+    DECREF's an array object which may have the (deprecated)
+    :c:data:`NPY_ARRAY_UPDATEIFCOPY` or :c:data:`NPY_ARRAY_WRITEBACKIFCOPY`
+    flag set without causing the contents to be copied back into the
+    original array. Resets the :c:data:`NPY_ARRAY_WRITEABLE` flag on the base
+    object. This is useful for recovering from an error condition when
+    writeback semantics are used, but will lead to wrong results.
+
+
+Enumerated Types
+^^^^^^^^^^^^^^^^
+
+.. c:type:: NPY_SORTKIND
+
+    A special variable-type which can take on different values to indicate
+    the sorting algorithm being used.
+
+    .. c:var:: NPY_QUICKSORT
+
+    .. c:var:: NPY_HEAPSORT
+
+    .. c:var:: NPY_MERGESORT
+
+    .. c:var:: NPY_STABLESORT
+
+        Used as an alias of :c:data:`NPY_MERGESORT` and vica versa.
+
+    .. c:var:: NPY_NSORTS
+
+       Defined to be the number of sorts. It is fixed at three by the need for
+       backwards compatibility, and consequently :c:data:`NPY_MERGESORT` and
+       :c:data:`NPY_STABLESORT` are aliased to each other and may refer to one
+       of several stable sorting algorithms depending on the data type.
+
+
+.. c:type:: NPY_SCALARKIND
+
+    A special variable type indicating the number of "kinds" of
+    scalars distinguished in determining scalar-coercion rules. This
+    variable can take on the values :c:data:`NPY_{KIND}` where ``{KIND}`` can be
+
+        **NOSCALAR**, **BOOL_SCALAR**, **INTPOS_SCALAR**,
+        **INTNEG_SCALAR**, **FLOAT_SCALAR**, **COMPLEX_SCALAR**,
+        **OBJECT_SCALAR**
+
+    .. c:var:: NPY_NSCALARKINDS
+
+       Defined to be the number of scalar kinds
+       (not including :c:data:`NPY_NOSCALAR`).
+
+.. c:type:: NPY_ORDER
+
+    An enumeration type indicating the element order that an array should be
+    interpreted in. When a brand new array is created, generally
+    only **NPY_CORDER** and **NPY_FORTRANORDER** are used, whereas
+    when one or more inputs are provided, the order can be based on them.
+
+    .. c:var:: NPY_ANYORDER
+
+        Fortran order if all the inputs are Fortran, C otherwise.
+
+    .. c:var:: NPY_CORDER
+
+        C order.
+
+    .. c:var:: NPY_FORTRANORDER
+
+        Fortran order.
+
+    .. c:var:: NPY_KEEPORDER
+
+        An order as close to the order of the inputs as possible, even
+        if the input is in neither C nor Fortran order.
+
+.. c:type:: NPY_CLIPMODE
+
+    A variable type indicating the kind of clipping that should be
+    applied in certain functions.
+
+    .. c:var:: NPY_RAISE
+
+        The default for most operations, raises an exception if an index
+        is out of bounds.
+
+    .. c:var:: NPY_CLIP
+
+        Clips an index to the valid range if it is out of bounds.
+
+    .. c:var:: NPY_WRAP
+
+        Wraps an index to the valid range if it is out of bounds.
+
```

Comment:
`NPY_RAISE`, `NPY_CLIP`, `NPY_WRAP` are defined both here and in `array-method-and-functions.rst` in the documentation of `PyArray_Choose`. I think they belong here, `PyArray_Choose` should link to the the definition of `NPY_CLIPMODE`.